corehq/form_processor/submission_post.py

```diff
@@ -665,6 +665,8 @@ def _get_form_name_info(domain, build_id):
     """
     :return: (default_language, {'http://my.xmlns': {'en': 'My Form'}})
     """
+    if not build_id:
+        return 'en', {}
```

Comment:
Having this in here will still result in cache hits which seems unnecessary since this is a constant. If we did the check prior to calling this method it would bypass that and potentially other code.

Reply:
good call - moving this to the caller